mmselfsup/models/utils/extract_process.py

```diff
@@ -7,8 +8,40 @@
 
 
 class ExtractProcess(object):
-    """Extraction process for `extract.py` and `tsne_visualization.py` in
-    tools.
+    """Global average-pooled feature extraction process."""
+
+    def __init__(self):
+        self.avg_pool = nn.AdaptiveAvgPool2d((1, 1))
+
+    def _forward_func(self, model, **x):
+        """The forward function of extract process."""
+        backbone_feat = model(mode='extract', **x)
+        pooling_feat = self.avg_pool(backbone_feat[-1])
+        flat_feat = pooling_feat.view(pooling_feat.size(0), -1)
+        return dict(feat=flat_feat.cpu())
+
+    def extract(self, model, data_loader, distributed=False):
+        """The extract function to apply forward function and choose
+        distributed or not."""
+        model.eval()
+
+        # the function sent to collect function
+        def func(**x):
+            return self._forward_func(model, **x)
+
+        if distributed:
+            rank, world_size = get_dist_info()
+            results = dist_forward_collect(func, data_loader, rank,
+                                           len(data_loader.dataset))
+        else:
+            results = nondist_forward_collect(func, data_loader,
+                                              len(data_loader.dataset))
+        return results
+
+
+class MultiExtractProcess(object):
```

Comment:
Why split original `ExtractProcess` into `MultiExtractProcess` and `ExtractProcess`